edx_rest_framework_extensions/permissions.py

```diff
@@ -1,9 +1,71 @@
 """ Permission classes. """
+from oauth2_provider.ext.rest_framework.permissions import TokenHasScope
+from opaque_keys.edx.keys import CourseKey
+from rest_framework.exceptions import PermissionDenied
 from rest_framework.permissions import BasePermission
 
+from edx_rest_framework_extensions.decorators import (
+    jwt_authentication_required,
+    skip_unless_jwt_scopes_enforced,
+)
+from edx_rest_framework_extensions.utils import jwt_decode_handler
+
 
 class IsSuperuser(BasePermission):
     """ Allows access only to superusers. """
 
     def has_permission(self, request, view):
         return request.user and request.user.is_superuser
+
+
+class JwtHasScope(TokenHasScope):
+    """
+    The request is authenticated as a user and the token used has the right scope.
+    """
+
+    @skip_unless_jwt_scopes_enforced
+    @jwt_authentication_required
+    def has_permission(self, request, view):
+        jwt_scopes = jwt_decode_handler(request.auth).get('scopes', [])
+        required_scopes = set(self.get_scopes(request, view))
+        if required_scopes.issubset(jwt_scopes):
+            return True
+
+        if required_scopes:
+            # pylint: disable=attribute-defined-outside-init
```

Comment:
Awesome, thanks! I'll check to see if I can specify a dict doing it that way.

Reply:
So, specifying the message as a class variable with work with the dict. The problem doing it that way though is that I wouldn't be able to specify the required scopes in the message because those are defined on the view class. I'm going to leave it this way both here and below for now.

Reply:
I added a comment below about adding a test case for this.  I don't see any code in StackOverflow or in the `Permission` classes where they set request-specific `message` values.  Also, is there any downstream code that assumes `message` is a string?

So I'm unclear whether this way is supported.

Reply:
It is supported, see this DRF code: https://github.com/encode/django-rest-framework/blob/master/rest_framework/exceptions.py#L56. I'll see if I can add a test case for the message.

Reply:
On second thought, even though this will work it seems weird to be modifying a class variable from an instance method. I'm going to stick with a generic missing required scopes/filter message defined at the class level.